src/database.ts

```diff
@@ -182,17 +283,25 @@ class Database extends ServiceObject {
    *   readTimestamp: 1518464696657
    * });
    */
-  batchTransaction(identifier, options?) {
-    let session = identifier.session;
+  batchTransaction(
+    identifier: TransactionIdentifier,
+    options?: TimestampBounds
+  ): BatchTransaction {
+    const session =
+      typeof identifier.session === 'string'
+        ? this.session(identifier.session)
+        : identifier.session;
     const id = identifier.transaction;
-    if (is.string(session)) {
-      session = this.session(session);
-    }
-    const transaction = new BatchTransaction(session, options);
+    const transaction: BatchTransaction = new BatchTransaction(
+      session,
+      options
+    );
     transaction.id = id;
-    transaction.readTimestamp = identifier.readTimestamp;
+    transaction.readTimestamp = identifier.readTimestamp as PreciseDate;
```

Comment:
I think you've stumbled upon a bug as this does not match the return type for `BatchTransaction#identifier`. I would either update this to be `timestamp` or update `identifier()` to return `readTimestamp`.

Reply:
looks breaking changes to rename in identifier(), so renamed in TransactionIdentifier Property.